custom_components/tahoma/entity.py

```diff
@@ -71,56 +65,24 @@ def device_info(self) -> DeviceInfo:
                 OverkizState.CORE_PRODUCT_MODEL_NAME,
                 OverkizState.IO_MODEL,
             )
-            or self.device.widget
+            or self.device.widget.value
         )
 
         return DeviceInfo(
             identifiers={(DOMAIN, self.executor.base_device_url)},
             name=self.device.label,
-            manufacturer=manufacturer,
-            model=model,
-            sw_version=self.executor.select_attribute(
-                OverkizAttribute.CORE_FIRMWARE_REVISION
+            manufacturer=str(manufacturer),
+            model=str(model),
+            sw_version=cast(
+                str,
+                self.executor.select_attribute(OverkizAttribute.CORE_FIRMWARE_REVISION),
             ),
+            hw_version=self.device.controllable_name,
             suggested_area=self.coordinator.areas[self.device.place_oid],
-            via_device=self.executor.get_gateway_id(),
+            via_device=(DOMAIN, self.executor.get_gateway_id()),
             configuration_url=self.coordinator.client.server.configuration_url,
         )
 
-    @property
-    def extra_state_attributes(self) -> dict[str, Any]:
```

Comment:
We will have to add this as breaking change.

Reply:
yeah, there are more. I will add some text to the changelog! :)